frame/support/src/lib.rs

```diff
@@ -946,3 +949,965 @@ mod tests {
 		})
 	}
 }
+
+/// prelude to be used alongside pallet macro, for ease of use.
+pub mod pallet_prelude {
+	pub use sp_std::marker::PhantomData;
+	pub use frame_support::traits::{Get, Interface, IsType, GetPalletVersion};
+	#[cfg(feature = "std")]
+	pub use frame_support::traits::GenesisBuild;
+	pub use frame_support::dispatch::{DispatchResultWithPostInfo, Parameter};
```

Comment:
```suggestion
	pub use frame_support::dispatch::{DispatchError, DispatchResult, DispatchResultWithPostInfo, Parameter};
```

Reply:
if we accept DispatchResult as output of dispatchable function then it would make sense to add them, but otherwise I think they shouldn't be used anymore.

Reply:
`DispatchError` is useful for defining custom functions with a return value:
```Rust
fn my_fallible_fn(param: u32) -> Result<u32, DispatchError> {
    // snip
}
```

Reply:
yes ok, I added DispatchError